### ISSUE TITLE
[BUG] Memory leak in SStream::str()

### ISSUE THREAD

**Issue by mallardtheduck** [blueloveTH/pocketpy#179]
### Bug description
I believe I've discovered a memory leak in the `SStream::str()` method.
The memory is released from `buffer.detach();` then passed into the Str constructor, which makes a copy of the string data, then the memory goes out-of-scope without being freed. I assume the intent was that the Str object take ownership the existing memory buffer, but this is not what happens.

### Steps to reproduce
A simple `auto vm = new pkpy::VM(true);` followed by `delete vm;` is enough to leak memory (apparently all the source codes strings used by the built-in modules). I wrote a program that just did that 1,000 times in a loop and used the MacOS/Xcode "leaks" tool to find the issue. Presumably valgrind or any other similar tool would find the same thing.

### Environment information
You need to provide these strings:
- pocketpy's version: 1.3.1
- platform string (win32/linux/darwin/emscripten/android/ios): MacOS 14.1.1 (Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:27 PDT 2023; root:xnu-10002.41.9~6/RELEASE_X86_64) Although the issue isn't platform-specific.
- 32-bit or 64-bit: 64-bit

### Additional context
My fix was to add an additional `Str` constructor that takes a `std::pair<char*, int>` (the return value from `.detach()`) directly and takes ownership of the buffer and modify the `SStream::str()` method appropriately.

**Comment by blueloveTH** [comment]
Thanks for your report. I will check it as soon as possible.
